src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.cs

```diff
@@ -148,24 +154,23 @@ private async Task ProduceTagsAsync(TaggerContext<TTag> context, DocumentSnapsho
             var suppressedDiagnosticsSpans = (NormalizedSnapshotSpanCollection?)null;
             buffer?.Properties.TryGetProperty(PredefinedPreviewTaggerKeys.SuppressDiagnosticsSpansKey, out suppressedDiagnosticsSpans);
 
+            var diagnosticCacheService = workspace.Services.GetService<IDiagnosticCacheService>();
+            if (diagnosticCacheService != null)
+            {
+                // During solution load. if there's any cached diagnostics available,
+                // this service will push updates via CachedDiagnosticsUpdated event.
+                await diagnosticCacheService.LoadCachedDiagnosticsAsync(document, cancellationToken).ConfigureAwait(false);
+
+                if (diagnosticCacheService.TryGetLoadedCachedDiagnostics(document.Id, out var cachedId, out var cachedDiagnostics))
```

Comment:
Seems little weird to invoke Load method and then a TryGetLoaded method. Maybe better to use an additional parameter `bool loadCachedDiagnostics` for `TryGetLoadedCachedDiagnostics` method?